packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -203,6 +204,9 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
     private _baseSnapshot: ISnapshotTree | undefined;
     protected _attachState: AttachState;
     protected readonly summarizerNode: ISummarizerNodeWithGC;
+    public readonly logger: ITelemetryLogger;
+    private readonly thresholdOpsCounter: ThresholdCounter;
+    private static readonly pendingOpsCountThreshold = 1000;
```

Comment:
1K ops is about 0.5Mb in memory, and (on slow device) is about 1s to process.
We probably want to know a bit sooner, maybe 500 or 200? We can always increase it if we learn cost of telemetry is too high.

Reply:
will do

Reply:
I would expect collectors to want to perform sampling based on event criteria more than they would want to request that code not be instrumented. As a separate concern oriented more around the framework than this pull -  is there any discussion around aligning the framework to an open standard in this space? (or is it already?)

Reply:
I'm not sure I follow what you propose RE sampling. Are you saying that some external entity should control sampling rates?

> open standard in this space

You mean telemetry (interfaces)? Or something else?

Reply:
I am fishing for what the relationship between Fluid Framework and [OpenTelemetry](https://cloudblogs.microsoft.com/opensource/2019/05/23/announcing-opentelemetry-cncf-merged-opencensus-opentracing/) is or should be. 

I am in the process of doing some catch up in the space myself though.

Reply:
Never heard about it :) We should definitely look into it - feel free to ping Mark,
That said, this smells a bit:

> You can get a tracer by calling getTracer on the global tracer provider. 

Globals are almost always bad starting point